src/main/java/edu/cmu/cs/cs214/rec10/plugin/SortWithSwapsPlugin.java

```diff
@@ -0,0 +1,118 @@
+package edu.cmu.cs.cs214.rec10.plugin;
+
+import edu.cmu.cs.cs214.rec10.framework.core.GameFramework;
+import edu.cmu.cs.cs214.rec10.framework.core.GamePlugin;
+
+import java.util.Random;
+
+public class SortWithSwapsPlugin implements GamePlugin {
+    private int ONE = 1;
+    private int TEN = 10;
+
+    // Describes whether the current move is over
+    public Boolean move = false;
+
+    // An array of the numbers 1-10
+    private int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+    // The game framework
+    private GameFramework framework = null;
+
+    // The last number clicked, which is to be swapped with the next number clicked
+    private int last = -1;
+
+    //The number of swaps the user has made
+    static Integer count = null;
+
+    public String getGameName() {
+        return new String("sort with swaps plugin");
+    }
+
+    public int getGridWidth() {
+        return TEN;
+    }
+
+    public int getGridHeight() {
+        return ONE;
+    }
+
+    public void onRegister(GameFramework framework) {
+        this.framework = framework;
+    }
+
```

Comment:
Code uses descriptive and meaningful variable, method and class names. Code understanding
does not rely too much on comments.